parking_permits/models/parking_permit.py

```diff
@@ -387,6 +387,10 @@ def zone_changed(self):
             address and address.zone == self.parking_zone for address in addresses
         )
 
+    def current_period_end_time_with_fixed_months(self, months):
+        end_time = get_end_time(self.start_time, months)
+        return max(self.start_time, end_time)
+
```

Comment:
It seems the `current_period_end_time` property can use this method:

```python
    @property
    def current_period_end_time(self):
        return self.current_period_end_time_with_fixed_months(self.months_used)

```

Reply:
Good point. I'll update this still.